### ISSUE TITLE
How to implement baritone Api to forge mod?

### ISSUE THREAD

**Issue by Philip2909** [cabaletta/baritone#2040]
## What do you need help with?
How to implement baritone Api to forge mod? I am new to forge modding and java.

## Final checklist
- [x] I know how to properly use check boxes
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by l1ving** [comment]
Read the readme, or you can just look at what we did [here](https://github.com/kami-blue/client/commits/master/build.gradle) in these commits, in order

- https://github.com/kami-blue/client/commit/4d57832e23ce22cfd01292e45ff9b22d5ab9b337#diff-c197962302397baf3a4cc36463dce5ea
- https://github.com/kami-blue/client/commit/5985ee42f1d8f7bbed035163f11b841840407ca6#diff-c197962302397baf3a4cc36463dce5ea
- https://github.com/kami-blue/client/commit/1de50c8865e33e08f88d73a6f7717f42cff86cb5#diff-c197962302397baf3a4cc36463dce5ea
- https://github.com/kami-blue/client/commit/34cbdb813c9085e2655dc5581f7503a9111eed84
- https://github.com/kami-blue/client/commit/8da73b902874a25b2c0f74fc528f6854fc879c9f

You can probably combine those into 1 visible commit but I'm lazy atm

**Comment by Philip2909** [comment]
Hi,
I read the Readme but it just included what commands you can add when you have imported the mod.

My build.gradle looks completely different, and when I add the compile stuff I just get errors that it can't find the files on the forgemodding site. Is this becuase it's a new version? I even tried to put the api jar in the folder it told me.

Here is the error:
Could not resolve all files for configuration ':compileClasspath'.
> Could not find cabaletta:baritone-api:1.6.1.
  Searched in the following locations:
    - https://files.minecraftforge.net/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://files.minecraftforge.net/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - file:/D:/Users/Philip/.gradle/caches/forge_gradle/bundeled_repo/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - file:/D:/Users/Philip/.gradle/caches/forge_gradle/bundeled_repo/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://libraries.minecraft.net/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
  Required by:
      project :

Possible solution:
 - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html


Any way to solve this, if so how?

**Comment by l1ving** [comment]
You forgot to add the impact maven repo to your deps

> My build.gradle looks completely different

Well no shit, you're supposed to reimplement the *diffs*  which I presented (in green!!)

**Comment by Philip2909** [comment]
Ok, I added they maven but I don't understand what you mean by reimplementing diffs? Whenever i use "shadowJar" it crashes immediately. The error after I added everything I could that was green is:

Could not resolve all files for configuration ':compileClasspath'.
> Could not find cabaletta:baritone-api:1.6.1.
  Searched in the following locations:
    - http://repo.spongepowered.org/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - http://repo.spongepowered.org/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://impactdevelopment.github.io/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://impactdevelopment.github.io/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - http://maven-eclipse.github.io/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - http://maven-eclipse.github.io/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://jitpack.io/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://jitpack.io/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://jcenter.bintray.com/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://jcenter.bintray.com/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://files.minecraftforge.net/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://files.minecraftforge.net/maven/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - file:/D:/Users/Philip/.gradle/caches/forge_gradle/bundeled_repo/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - file:/D:/Users/Philip/.gradle/caches/forge_gradle/bundeled_repo/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://libraries.minecraft.net/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.pom
    - https://repo.maven.apache.org/maven2/cabaletta/baritone-api/1.6.1/baritone-api-1.6.1.jar
  Required by:
      project :

Possible solution:
 - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html

What I am doing wrong?

**Comment by l1ving** [comment]
You should be doing 1.6, not 1.6.1

**Comment by l1ving** [comment]
![image](https://user-images.githubusercontent.com/17222512/94373326-bbdeb900-00d2-11eb-95cc-ab978d5fe758.png)
I specifically linked this for a reason

**Comment by l1ving** [comment]
You compile the 1.6 api but the 1.6.1 baritone

**Comment by Philip2909** [comment]
Ok thanks I got it running but how do I add the baritoneAPI.getProvider()...etc

**Comment by l1ving** [comment]
What?

**Comment by Philip2909** [comment]
I got it running the game but how do I import to be able to do this:
BaritoneAPI.getSettings().allowSprint.value = true;
BaritoneAPI.getSettings().primaryTimeoutMS.value = 2000L;

**Comment by l1ving** [comment]
Import the correct class

**Comment by Philip2909** [comment]
import baritone.api.BaritoneAPI;
does not work for me

**Comment by Philip2909** [comment]
did I forget someting?

**Comment by l1ving** [comment]
You need to reimport as it's probably not in your classpath

**Comment by Philip2909** [comment]
sorry that i am an idiot how do I do that?

**Comment by Philip2909** [comment]
dumb qustion, where is the path to the baritone-api.jar?

**Comment by Philip2909** [comment]
is it a problem that I am using Intellij?

**Comment by l1ving** [comment]
No. 

> where is the path to the baritone-api.jar?

Why??

**Comment by l1ving** [comment]
> sorry that i am an idiot how do I do that?

Press :arrows_counterclockwise:

**Comment by l1ving** [comment]
![image](https://user-images.githubusercontent.com/17222512/94374710-25af9080-00dc-11eb-8234-64125146199f.png)

**Comment by Philip2909** [comment]
Found the refresh button, it was hidden in some tab to the left. Thanks for helping me out!

**Comment by Philip2909** [comment]
Sorry to bother you, ran into some problems.
When I run it with both baritone and baritone api it says Exception in thread "main"
When I run it with only the api I get a mixin error, I have never done anyting in a mixin, is that someting I need to fix?

**Comment by l1ving** [comment]
Yes, you need to register the Baritone mixins

**Comment by Philip2909** [comment]
How and where do I do that? I added the 'MixinConfigs':  mixins.baritone.json' to the gradle? And when I run it sould I use only the API or both?

**Comment by l1ving** [comment]
You need the forge mixin whatever. I linked commits to the example

**Comment by Philip2909** [comment]
I can't get it to work, it does not find the IFMLLoadingPlugin

**Comment by l1ving** [comment]
You need to add it to your mixin loader

**Comment by Philip2909** [comment]
I have no idea how to do that and when I try to find info about it it's about fabric and events. Is it with the @Mixin thing?

**Comment by l1ving** [comment]
Heres mine: https://github.com/kami-blue/client/blob/master/src/main/java/me/zeroeightsix/kami/mixin/MixinLoaderForge.java

**Comment by Philip2909** [comment]
I have found that one and tried to make is as good as I could but it just can't find "IFMLLoadingPlugin" And all my @Overrides turn red i  have honestly no idea why

**Comment by l1ving** [comment]
You need to reimport your project then

**Comment by Philip2909** [comment]
![image](https://user-images.githubusercontent.com/51290462/94467873-d6dd2600-01c3-11eb-94e2-6a9fd4dcecaa.png)
I have reimported and reloaded the project, could it have been renamed?

**Comment by l1ving** [comment]
If you updated it, yes. Don't write imports manually.

**Comment by Philip2909** [comment]
I closed the project the cleared the cashce and that started to import some things when I reopned the project. I Usally don't write them manually but I wanted to how that there is nothing on that path.

**Comment by Philip2909** [comment]
Do you have any idea what I sould do? Trying to find if it has a new name

**Comment by l1ving** [comment]
Let your IDE handle your imports for you

**Comment by Philip2909** [comment]
![image](https://user-images.githubusercontent.com/51290462/94472321-ddbb6700-01ca-11eb-8aaa-b93119ce0f25.png)
It finds the path, net.minecraftforge.fml.relauncher.IFMLLoadingPlugin but it still is red
Could the name be "FMLLOADER"? And is this IFMLLoadingPlugin a must have?

**Comment by l1ving** [comment]
You probably have the wrong name

**Comment by Philip2909** [comment]
Do you have any ideas of how to find the name of it?

**Comment by Philip2909** [comment]
What happens if you done use IFMLLoadingPlugin?

**Comment by l1ving** [comment]
> Do you have any ideas of how to find the name of it?
Idk guess lmao

> What happens if you done use IFMLLoadingPlugin?
Baritone won't load it's Mixins if you want to include Baritone

**Comment by Philip2909** [comment]
Hi, I tried to ask on the maker of that package, on minecraft forge modding support but there were rude and didn't wanna help me, so I guess I will have to figure it out somehow. Btw I am allowed to clone that client project you did and see what you did different and if I have missed something?

**Comment by l1ving** [comment]
I mean yea. It's LGPLv3

**Comment by bebeli555** [comment]
He was trying to add it to a 1.16 mod not 1.12.2
And theres no IFMLLoadingPlugin in 1.16
